homeassistant/components/zwave_mqtt/climate.py

```diff
@@ -0,0 +1,390 @@
+            return None
+        return self._current_mode_setpoint_values[0].value
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the lowbound target temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) < 2:
+            return None
+        return self._current_mode_setpoint_values[0].value
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the highbound target temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) < 2:
+            return None
+        return self._current_mode_setpoint_values[1].value
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature.
+
+        Must know if single or double setpoint.
+        """
+        if len(self._current_mode_setpoint_values) == 1:
+            setpoint = self._current_mode_setpoint_values[0]
+            target_temp = kwargs.get(ATTR_TEMPERATURE)
+            if setpoint is not None and target_temp is not None:
+                setpoint.send_value(target_temp)
+        elif len(self._current_mode_setpoint_values) == 2:
+            (setpoint_low, setpoint_high) = self._current_mode_setpoint_values
+            target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+            target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+            if setpoint_low is not None and target_temp_low is not None:
+                setpoint_low.send_value(target_temp_low)
+            if setpoint_high is not None and target_temp_high is not None:
+                setpoint_high.send_value(target_temp_high)
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new target fan mode."""
+        if not self.values.fan_mode:
+            return
+        # get id for this fan_mode
+        fan_mode_value = None
+        for val in self.values.fan_mode.value[VALUE_LIST]:
+            if val[VALUE_LABEL] == fan_mode:
+                fan_mode_value = val[VALUE_ID]
+        if fan_mode_value is None:
+            _LOGGER.warning("%s is an invalid fan_mode!", fan_mode)
+        else:
+            self.values.fan_mode.send_value(fan_mode_value)
+
+    async def async_set_swing_mode(self, swing_mode):
+        """Set new target swing mode."""
+        if not self.values.swing_mode:
+            return
+        # get id for this swing_mode
+        swing_mode_value = None
+        for val in self.values.swing_mode.value[VALUE_LIST]:
+            if val[VALUE_LABEL] == swing_mode:
+                swing_mode_value = val[VALUE_ID]
+        if swing_mode_value is None:
+            _LOGGER.warning("%s is an invalid swing_mode!", swing_mode)
+        else:
+            self.values.swing_mode.send_value(swing_mode_value)
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        if not self.values.mode:
+            return
+        hvac_mode_value = HVAC_MODE_ZW_MAPPINGS[hvac_mode]
```

Comment:
Should we use dict.get here if we receive a mode that we haven't mapped?
```suggestion
        hvac_mode_value = HVAC_MODE_ZW_MAPPINGS.get(hvac_mode)
        if hvac_mode_value is None:
            _LOGGER.warning("Received an invalid hvac mode %s", hvac_mode)
            return
```

Reply:
Yeah, good catch!